### ISSUE TITLE
Double scrollbar in some conversations, makes wheel-scrolling more difficult

### ISSUE THREAD

**Issue by squelart** [thunderbird-conversations/thunderbird-conversations#1517]
* **Thunderbird version:** 79.0b3 (32-bit)
* **Thunderbird Conversation version:** 3.1.19
* **Operating System:** Windows 10.0.18363 Build 18363

**What did you do?** Just reading emails that are taller than the message window.

**What did you expect to happen?** One vertical scrollbar.

**What actually happened?** Two vertical scrollbars!
It looks like there is a scrollbar inside the message itself, though it can only move 1 (or a very small number) pixel up or down, so I'm guessing somehow Conversations doesn't get the correct size. Off-by-1, or floating-point computation somewhere? 🤔

Because of this inner scrollbar, the mouse-wheel first works on it, so I need to stop the wheel and restart scrolling in order to influence the outer scrollbar and really scroll to see more of the message, it's quite annoying, especially when I need to scroll up and down a long message.

Snapshot of the top-right corner of the message, showing the two scrollbars.
![image](https://user-images.githubusercontent.com/8499548/88998533-9bde8380-d335-11ea-8e6e-c2173a68fd03.png)

It's only happening with a subset of single messages, never conversations with 2 or more emails.
And sometimes I see the two scrollbars for a second, and then the inner one disappears.
Conversations is my only add-on, and I don't see this problem if I disable it.

**Comment by Standard8** [comment]
Yeah, we already do various things to try and avoid this and already have a workaround in place: https://github.com/thunderbird-conversations/thunderbird-conversations/blob/c1a50cca2b9df461ff455998307fedf253b81bbb/addon/content/messageIFrame.jsx#L159

Though maybe I'll bump that to + 2 or + 3 and see how that goes.

**Comment by squelart** [comment]
Thank you @Standard8 for this.

Has this landed in Nightly by now? I've got 80.0b5 (32-bit) build id 20200818225932
I'm still seeing some double scrollbars, but I'm using the "Thunderbird Conversations" add-on; disabling the add-on removes the double scrollbar (in the few emails I've tried). Should I raise the issue over at the add-on's site, or is that a potential Thunderbird issue, or even a combination?

**Comment by Standard8** [comment]
> Has this landed in Nightly by now? I've got 80.0b5 (32-bit) build id 20200818225932

The attempted fix was in conversations.

> I'm still seeing some double scrollbars, but I'm using the "Thunderbird Conversations" add-on; disabling the add-on removes the double scrollbar (in the few emails I've tried). Should I raise the issue over at the add-on's site, or is that a potential Thunderbird issue, or even a combination?

This is the Conversations site.

It does still seem to be an issue, and having looked around I think it is down to an issue in the core of Thunderbird: https://bugzilla.mozilla.org/show_bug.cgi?id=1247608

I'm thinking there might be a way around it, but I haven't had time to investigate in detail yet.

**Comment by squelart** [comment]
> This is the Conversations site.
Woops, sorry for the confusion 😅
Thank you for looking into this.

**Comment by zsunsun** [comment]
This issue has been open for a while and driving me absolutely bonkers.  Any news on alternate workarounds?

**Comment by Standard8** [comment]
I think I've just identified the cause of the issue. I think fundamentally the fault is emails that are badly structured, but this could be an issue with the html standards or with the back end of Thunderbird.

From the cases I can find in my inboxes, this appears to happen when the email has a table element with height set to "100%".

In that case, the height of the table is set to the height of the browser it is contained within.

Unfortunately, in these cases, you also have items outside the table, e.g.:

```
-----
Some text = 15 pixels height
-----
Table = 100% height
-----
```

So, if you set the height of the containing element to say, 500 pixels, the table will now be 500 pixels in height, but the additional text will make the whole thing 515 pixels in height.

If you increased the height of the container, then the table would increase height again, and so you're stuck in this always-larger-than-container case.

As I mentioned, I'm not quite sure where the root cause lies. In theory the emails are being constructed badly, but I doubt we're going to get the providers to fix them properly.

Note, I have just made a small change (23cb708ad38c84e6f6499bc3761b61e20a9782b1) which may improve the odd case, but if the email is using tables, there's unfortunately not much to be done about it.

**Comment by Standard8** [comment]
I just realised, you can also see this effect in Thunderbird's email reader.

Open the email in a new tab (to give you more vertical height), then zoom out so that the email should be shorter than the view. You'll still be able to scroll it.

If you switch to an email that doesn't use 100% table height, then you won't be able to scroll that.
